### DISCUSSION TITLE
🎉 JointJS+ v4.1 is here: The most feature-rich release so far

### DISCUSSION THREAD

**Discussion by kumilingus** [clientIO/joint#2825] [News]
The latest release brings a host of new features and enhancements to _JointJS+_, including new applications, components, and updates to existing ones.

Key highlights of this release include:
- **Five new applications**: [BPMNPools](https://docs.jointjs.com/learn/release-notes/4.1.0#apps-BPMNPools), [SheetCutting](https://docs.jointjs.com/learn/release-notes/4.1.0#apps-SheetCutting), [TheoryOfChange](https://docs.jointjs.com/learn/release-notes/4.1.0#apps-TheoryOfChange), [TreeDesigner](https://docs.jointjs.com/learn/release-notes/4.1.0#apps-TreeDesigner), and a redesigned [KitchenSink](https://docs.jointjs.com/learn/release-notes/4.1.0#apps-KitchenSink).
Refactored [ForceDirectedGraph](https://docs.jointjs.com/learn/release-notes/4.1.0#forcedirected) layout for improved performance and usability.
- **New [Halo](https://docs.jointjs.com/learn/release-notes/4.1.0#halo) type** that allows you to customize and organize handles as desired.
- **Support for different cell selection methods and their visualization**, by leveraging new [selection frames](https://docs.jointjs.com/learn/release-notes/4.1.0#ui-Selection-frames) and [selection regions](https://docs.jointjs.com/learn/release-notes/4.1.0#ui-Selection-regions).
- **Optimized spatial queries**, with the introduction of the new [SearchGraph](https://docs.jointjs.com/learn/release-notes/4.1.0#searchgraph) component.
- **Extendable SVG language** that allows you to define shapes with [custom SVG attributes](https://docs.jointjs.com/learn/release-notes/4.1.0#dia-CellView-specialPresentationAttributes), enabling more advanced visualizations.
Explore more highlights in our [official announcement](https://www.jointjs.com/blog/introducing-4-1), or [read the full release notes](https://docs.jointjs.com/learn/release-notes/4.1.0) for more technical details.

Happy diagramming! :raised_hands:

**Comment by frnora** [comment]
Congratulations on the new version, looking forward to trying out the new features!

(not quite) side note: The docs search functionality doesn't work for me at all for 4.1, 4.0 (still) works without problems.
<img width="691" alt="4 1" src="https://github.com/user-attachments/assets/7c0b9af4-0e98-47d9-9cac-9b0c584ccea3">

**Reply by kumilingus** [reply]
The search should now be fully functional. Thank you!
